### PR TITLE
More tests for the commandline interface

### PR DESCRIPTION
**Pull request by chStaiger**
I added more tests:

- Check if columns only consist of NaN
- Check if clumns adhere to their categorical values defined in the validation file in the section `values`
- A check that tests whether each row in a column cluster does not contain single NaN values.

I also added some hopefully convenient tool that you get asked after each section whether you want to continue the tests. This can be overridden by the option `-y` which will enforce all tests.

For doing all tests:
```
python3 src/validate.py -d data/example_study1 -y
```

For intercepting after some tests:
```
python3 src/validate.py -d data/example_study1
```

### CODE REVIEW

File: src/validation_utils.py [hunk 1]
```diff
@@ -94,12 +108,76 @@ def check_column_exists(data: dict, validation: dict) -> list:
...
+                else:
+                    unexpected_values.append(
+                        (data_name, var_name, check_categorical_values(col, cat_values))
+                    )
+    return unexpected_values
+
+
+def check_column_clusters(cols: dict, data_frame: pd.DataFrame) -> list:
+    """Check if row contains values in all columns of the cluster.
+    Return row and col names if not.
```
Comment (qubixes) [inline src/validation_utils.py:180]:
```suggestion
    """Check if row contains values in all columns of the cluster.

    Return row and col names if not.
```

File: src/validation_utils.py [hunk 2]
```diff
@@ -136,24 +217,28 @@ def identifier_checks(data: dict, id_col_names: list) -> bool:
...
-    
+
     for col_name in id_col_names:
         for data_name in [d for d in data if col_name in data[d]]:
             if not set(data[data_name][col_name]).issubset(ids[col_name]):
-                msg = f"File {data_name} contains undefined ids in column {col_name}:\n"+ \
-                      f"{set(data[data_name][col_name]).difference(ids[col_name])}"
+                msg = (
+                    f"File {data_name} contains undefined ids in column {col_name}:\n"
+                    + f"{set(data[data_name][col_name]).difference(ids[col_name])}"
```
Comment (qubixes) [inline src/validation_utils.py:240]:
```suggestion
                    f"{set(data[data_name][col_name]).difference(ids[col_name])}"
```

**Review by qubixes — APPROVED** [review]
Very nice PR, don't really have much to add this time!

**Comment by lagerweijgr** [comment]
I have some questions/things about the changes: 

1. I get the error "←[91mERROR Empty column inoculation_type in host_events.csv←[0m" 
I think this is because there is no value in this column. So this is an exception for the empty columns. There is also a possibility that a experiment has no inoculation event. 

2. This error: "←[91mERROR VALUES environment.csv column environment_level contains {1} --> not accepted.←[0m"
I see in the validation.toml that 1,2,3 are accepted ? So why my value of 1 is not accepted? 

3. I get the feeling that the check on missing values per row is 1 step behind. I get 
Row 2 in columns ['measurement_type', 'measurement_quantity', 'measurement_unit'].
Row 3 in columns ['measurement_type', 'measurement_quantity', 'measurement_unit'].

where in the excel file of host_event.csv it is row 3-4 (if I do not count the header as a row ;)).

4. Maybe it is possible to show 5 more errors hence 15 errors instead of 10?

**Comment by chStaiger** [comment]
>     1. I get the error "←[91mERROR Empty column inoculation_type in host_events.csv←[0m"
>        I think this is because there is no value in this column. So this is an exception for the empty columns. There is also a possibility that a experiment has no inoculation event.

It can be easily turned into "INFO", but then it would also just give an INFO for the madatory columns.


>     2. This error: "←[91mERROR VALUES environment.csv column environment_level contains {1} --> not accepted.←[0m"
>        I see in the validation.toml that 1,2,3 are accepted ? So why my value of 1 is not accepted?
> 
I need to on which data you are running the code. For the synthetic data 2 all is fine.


>     3. I get the feeling that the check on missing values per row is 1 step behind. I get
>        Row 2 in columns ['measurement_type', 'measurement_quantity', 'measurement_unit'].
>        Row 3 in columns ['measurement_type', 'measurement_quantity', 'measurement_unit'].
> 
> 
> where in the excel file of host_event.csv it is row 3-4 (if I do not count the header as a row ;)).

Yes, python starts indexing at 0. So the first row always has index 0. I added a 1.

> 
>     4. Maybe it is possible to show 5 more errors hence 15 errors instead of 10?

I think this is easily adjustable, if you need it. The variable to steer the length of the output is called `MAX_NUM_RES`

**Pull request merged**